website/docs/r/dx_private_virtual_interface.html.markdown

```diff
@@ -0,0 +1,63 @@
+---
+layout: "aws"
+page_title: "AWS: aws_dx_private_virtual_interface"
+sidebar_current: "docs-aws-resource-dx-private-virtual-interface"
+description: |-
+  Provides a Direct Connect private virtual interface resource.
+---
+
+# aws_dx_private_virtual_interface
+
+Provides a Direct Connect private virtual interface resource.
+
+## Example Usage
+
+```hcl
+resource "aws_dx_private_virtual_interface" "foo" {
+  connection_id = "dxcon-zzzzzzzz"
+
+  name           = "vif-foo"
+  vlan           = 4094
+  address_family = "ipv4"
+  bgp_asn        = 65352
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `connection_id` - (Required) The ID of the Direct Connect connection (or LAG) on which to create the virtual interface.
+* `vpn_gateway_id` - (Optional) The ID of the [virtual private gateway](vpn_gateway.html) to which to connect the virtual interface.
+* `dx_gateway_id` - (Optional) The ID of the Direct Connect gateway to which to connect the virtual interface.
+* `name` - (Required) The name for the virtual interface.
+* `vlan` - (Required) The VLAN ID.
+* `bgp_asn` - (Required) The autonomous system (AS) number for Border Gateway Protocol (BGP) configuration.
+* `bgp_auth_key` - (Optional) The authentication key for BGP configuration.
+* `address_family` - (Required) The address family for the BGP peer. `ipv4 ` or `ipv6`.
```

Comment:
Minor nitpick: Can you organize these with `Required` at the top, alphabetical, then `Optional`, alphabetical. Thanks!

Reply:
👍